syntax = "proto3";
package xcoin.interchange;

message Block{
  optional int64 index = 1;
  optional string hash = 2;
  optional string previousHash = 3;
  optional int64 timestamp = 4;
  optional string data = 5;
  optional int64 difficulty= 6;
  optional int64 nonce = 7;
  optional int64 minterBalance = 8;
  optional string minterAddress = 9;
}

message Blockchain{
  repeated Block blocks = 1;
}